### PR TITLE
AB#7632 - Filter to Dispensed or Filled only

### PR DESCRIPTION
**Pull request by sslaws**
Filters Medication Statements to return only Dispensed or Filled items.

### CODE REVIEW

File: Apps/Medication/src/Services/RestMedicationStatementService.cs [hunk 1]
```diff
@@ -74,6 +74,8 @@ public async Task<HNMessage<List<MedicationStatement>>> GetMedicationStatements(
             HNMessage<List<MedicationStatement>> hnClientMedicationResult = await this.RetrieveMedicationStatements(hdid, protectiveWord).ConfigureAwait(true);
             if (hnClientMedicationResult.Result == HealthGateway.Common.Constants.ResultType.Sucess)
             {
+                // Filter the results to return only Dispensed or Filled prescriptions.
+                hnClientMedicationResult.Message = hnClientMedicationResult.Message.Where(rx => rx.PrescriptionStatus == 'D' || rx.PrescriptionStatus == 'F').ToList<MedicationStatement>();
```
Comment (FuriousLlama) [inline Apps/Medication/src/Services/RestMedicationStatementService.cs]: Instead of strings, lets make them constants

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_208)

File: Apps/Medication/src/Services/RestMedicationStatementService.cs (hunk 1 above)
Comment (sslaws) [reply Apps/Medication/src/Services/RestMedicationStatementService.cs]: Resolved

**Comment by healthgatewaybc** [comment]
Common Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=commonlibrary_pr_208)

**Comment by healthgatewaybc** [comment]
Patient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=patientservice_pr_208)

**Comment by healthgatewaybc** [comment]
WebClient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=webclient_pr_208)

**Pull request merged**